.NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs

```diff
@@ -32,6 +32,7 @@ public static class DateTimeDefinitions
       public const string ThisPrefixRegex = @"(?<order>ce(tte)?|au\s+cours+(du|de))\b";
       public const string RangePrefixRegex = @"(du|depuis|des?|entre)";
       public const string DayRegex = @"(?<day>(?:3[0-1]|[1-2]\d|0?[1-9])(e(r)?)?)(?=\b|t)";
+      public const string WrittenDayRegex = @"(?<day>un|deux|trois|quatre|cinq|six|sept|huit|neuf|dix|onze|douze|treize|quatorze|quinze|seize|dix-sept|dix-huit|dix-neuf|vingt-et-un|vingt et un|vingt|vingt deux|vingt-deux|vingt trois|vingt-trois|vingt quatre|vingt-quatre|vingt cinq|vingt-cinq|vingt six|vingt-six|vingt sept|vingt-sept|vingt huit|vingt-huit|vingt neuf|vingt-neuf|trente|trente et un|trente-et-un)";
```

Comment:
This regex can be optimized a little to avoid repetitions and backtracking. This can come in a separate PR.

Reply:
modified